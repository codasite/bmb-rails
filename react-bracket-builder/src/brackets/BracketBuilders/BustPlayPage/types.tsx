import { MatchTree } from '../../shared/models/MatchTree'
import { BracketMeta } from '../../shared/context/context'

export interface BustPlayBuilderProps {
  matchTree?: MatchTree
  setMatchTree?: (matchTree: MatchTree) => void
  bracketMeta?: BracketMeta
  setBracketMeta?: (bracketMeta: BracketMeta) => void
  processing?: boolean
  handleSubmit?: () => void
}
