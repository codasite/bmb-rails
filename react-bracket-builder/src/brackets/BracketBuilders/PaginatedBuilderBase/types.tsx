import { MatchTree } from '../../shared/models/MatchTree'
import { BracketMeta } from '../../shared/context/context'

export interface PaginatedBuilderProps {
  matchTree?: MatchTree
  setMatchTree?: (matchTree: MatchTree) => void
  bracketMeta?: BracketMeta
  setBracketMeta?: (bracketMeta: BracketMeta) => void
  processing?: boolean
  handleSubmit?: () => void
  StartPageComponent?: React.FC<StartPageProps>
  BracketPagesComponent: React.FC<BracketPagesProps>
  EndPageComponent: React.FC<EndPageProps>
}

export interface StartPageProps {
  matchTree?: MatchTree
  onStart: () => void
}

export interface BracketPagesProps {
  matchTree?: MatchTree
  setMatchTree?: (matchTree: MatchTree) => void
  onFinished: () => void
}

export interface EndPageProps {
  matchTree?: MatchTree
  bracketMeta?: BracketMeta
  setBracketMeta?: (bracketMeta: BracketMeta) => void
  processing?: boolean
  handleSubmit?: () => void
  onEditClick?: () => void
  canEdit?: boolean
}
