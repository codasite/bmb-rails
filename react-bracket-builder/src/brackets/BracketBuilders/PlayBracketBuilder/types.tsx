import { MatchTree } from '../../shared/models/MatchTree'
import { BracketMeta } from '../../shared/context/context'

export interface PlayBuilderProps {
  matchTree?: MatchTree
  setMatchTree?: (matchTree: MatchTree) => void
  darkMode?: boolean
  setDarkMode?: (darkMode: boolean) => void
  bracketMeta?: BracketMeta
  setBracketMeta?: (bracketMeta: BracketMeta) => void
  handleApparelClick?: () => Promise<void>
  handleSubmitPicksClick?: () => Promise<void>
  processing?: boolean
  canPlay?: boolean
  showRegisterModal?: boolean
  setShowRegisterModal?: (showRegisterModal: boolean) => void
}
