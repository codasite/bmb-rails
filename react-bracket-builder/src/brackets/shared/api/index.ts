export type {
  TeamRes,
  TeamReq,
  MatchRes,
  MatchReq,
  MatchPick,
  TeamRepr,
  MatchTreeRepr,
  MatchRepr,
  BracketRes,
  BracketReq,
  PlayRes,
  PlayReq,
} from './types/bracket'
export { bracketApi } from './bracketApi'
