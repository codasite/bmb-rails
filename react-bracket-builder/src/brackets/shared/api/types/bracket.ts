import { Nullable } from '../../../../utils/types'
import { WildcardPlacement } from '../../models/WildcardPlacement'

interface phpDate {
  date: string
  timezone_type: number
  timezone: string
}

export interface TeamRes {
  id?: number
  name: string
}

export interface TeamReq {
  name: string
}

export interface MatchRes {
  id?: number
  roundIndex: number
  matchIndex: number
  team1?: Nullable<TeamRes>
  team2?: Nullable<TeamRes>
}

export interface MatchReq {
  roundIndex: number
  matchIndex: number
  team1?: TeamReq
  team2?: TeamReq
}

export interface MatchPick {
  roundIndex: number
  matchIndex: number
  winningTeamId: number
}

export interface TeamRepr {
  id?: number
  name: string
}

export interface MatchTreeRepr {
  rounds: Nullable<MatchRepr>[][]
  wildcardPlacement?: WildcardPlacement
}
export interface MatchRepr {
  id?: number
  roundIndex: number
  matchIndex: number
  team1?: TeamRepr
  team2?: TeamRepr
  team1Wins?: boolean
  team2Wins?: boolean
}
interface PostBase {
  id: number
  title: string
  status: string
  author: number
  authorDisplayName: string
  publishedDate: phpDate
  thumbnailUrl?: string
  url?: string
}
export interface BracketReq {
  title: string
  month: string
  year: string
  numTeams: number
  status?: string
  wildcardPlacement: WildcardPlacement
  matches: MatchReq[]
  results?: MatchPick[]
  updateNotifyPlayers?: boolean
}
export interface BracketRes extends PostBase {
  month: string
  year: string
  numTeams: number
  wildcardPlacement: WildcardPlacement
  matches?: MatchRes[]
  isOpen?: boolean
  isPrintable?: boolean
  fee?: number
}
export interface PlayReq {
  id?: number
  bracketId?: number
  title?: string
  status?: string
  picks: MatchPick[]
  bustedId?: number
  generateImages?: boolean
}
export interface PlayRes extends PostBase {
  bracketId: number
  picks: MatchPick[]
  bracket?: BracketRes
  bustedId?: number
  bustedPlay?: PlayRes
  isBustable?: boolean
}
