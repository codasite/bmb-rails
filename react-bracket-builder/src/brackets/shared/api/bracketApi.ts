import { BracketReq, BracketRes, PlayReq, PlayRes } from './types/bracket'
import { PaymentIntentReq, PaymentIntentRes } from './types/stripe'
import { NotificationReq } from './types/notification'
import { RequestOptions, WpHttpClient } from './wpHttpClient'

declare var wpbb_app_obj: any
export class BracketApi {
  private bracketsPath: string = 'brackets'
  private notificationsPath: string = 'notifications'
  private playPath: string = 'plays'
  private client: WpHttpClient
  constructor() {
    if (typeof wpbb_app_obj !== 'undefined') {
      this.client = new WpHttpClient(wpbb_app_obj.rest_url, wpbb_app_obj.nonce)
    }
  }
  async removeNotification(notificationId: number): Promise<boolean> {
    const options: RequestOptions = { method: 'DELETE' }
    return await this.client.performRequest(
      `${this.notificationsPath}/${notificationId}`,
      options
    )
  }

  async createNotification(
    notification: Partial<NotificationReq>
  ): Promise<NotificationReq> {
    const options: RequestOptions = { method: 'POST', body: notification }
    return await this.client.performRequest(this.notificationsPath, options)
  }
  async createBracket(bracket: BracketReq): Promise<BracketRes> {
    const options: RequestOptions = { method: 'POST', body: bracket }
    return await this.client.performRequest(this.bracketsPath, options)
  }
  async updateBracket(
    bracketId: number,
    bracket: Partial<BracketReq>
  ): Promise<BracketRes> {
    const options: RequestOptions = { method: 'PATCH', body: bracket }
    return await this.client.performRequest(
      `${this.bracketsPath}/${bracketId}`,
      options
    )
  }
  async deleteBracket(bracketId: number): Promise<boolean> {
    const options: RequestOptions = { method: 'DELETE' }
    return await this.client.performRequest(
      `${this.bracketsPath}/${bracketId}`,
      options
    )
  }
  async createPlay(play: PlayReq): Promise<PlayRes> {
    const options: RequestOptions = { method: 'POST', body: play }
    return await this.client.performRequest(this.playPath, options)
  }
  async getPlay(playId: number): Promise<PlayRes> {
    return await this.client.performRequest(`${this.playPath}/${playId}`)
  }
  async generatePlayImages(playId: number): Promise<PlayRes> {
    const options: RequestOptions = { method: 'POST' }
    return await this.client.performRequest(
      `${this.playPath}/${playId}/generate-images`,
      options
    )
  }
  async createStripePaymentIntent(
    req: PaymentIntentReq
  ): Promise<PaymentIntentRes> {
    const options: RequestOptions = { method: 'POST', body: req }
    return await this.client.performRequest(`stripe/payment-intent`, options)
  }
  async getStripeOnboardingLink(): Promise<string> {
    const options: RequestOptions = { method: 'POST' }
    return await this.client.performRequest(`stripe/onboarding-link`, options)
  }
}

export const bracketApi = new BracketApi()
