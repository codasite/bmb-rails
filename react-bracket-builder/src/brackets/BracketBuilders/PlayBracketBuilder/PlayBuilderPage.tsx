import React, { useContext, useEffect, useState } from 'react'
import { bracketApi } from '../../shared/api/bracketApi'
import { Nullable } from '../../../utils/types'
import { MatchTree } from '../../shared/models/MatchTree'
import { BracketMeta } from '../../shared/context/context'
import {
  WithBracketMeta,
  WithMatchTree,
} from '../../shared/components/HigherOrder'
import { BracketRes, PlayReq, PlayRes } from '../../shared/api/types/bracket'
import { PaginatedPlayBuilder } from '../PaginatedPlayBuilder/PaginatedPlayBuilder'
import { PlayBuilder } from './PlayBuilder'
import { getBracketWidth } from '../../shared/components/Bracket/utils'
import { getNumRounds } from '../../shared/models/operations/GetNumRounds'
import { WithWindowDimensions } from '../../shared/components/HigherOrder/WithWindowDimensions'
import { WindowDimensionsContext } from '../../shared/context/WindowDimensionsContext'
import { PlayStorage } from '../../shared/storages/PlayStorage'
import SubmitPicksRegisterModal from './SubmitPicksRegisterModal'
import StripePaymentModal from './StripePaymentModal'
import { logger } from '../../../utils/Logger'

const PlayBuilderPage = (props: {
  bracketProductArchiveUrl: string
  myPlayHistoryUrl: string
  isUserLoggedIn: boolean
  bracketStylesheetUrl: string
  bracket?: BracketRes
  play?: PlayRes
  matchTree?: MatchTree
  setMatchTree?: (matchTree: MatchTree) => void
  bracketMeta?: BracketMeta
  setBracketMeta?: (bracketMeta: BracketMeta) => void
  userCanPlayBracketForFree?: boolean
  loginUrl: string
}) => {
  const {
    bracket,
    bracketProductArchiveUrl,
    myPlayHistoryUrl,
    isUserLoggedIn,
    matchTree,
    setMatchTree,
    bracketMeta,
    setBracketMeta,
    userCanPlayBracketForFree,
  } = props

  const [processingAddToApparel, setProcessingAddToApparel] = useState(false)
  const [addToApparelError, setAddToApparelError] = useState(false)
  const [submitPicksError, setSubmitPicksError] = useState(false)
  const [processingSubmitPicks, setProcessingSubmitPicks] = useState(false)
  const [storedPlay, setStoredPlay] = useState<Nullable<PlayReq>>(null)
  const [showRegisterModal, setShowRegisterModal] = useState(!isUserLoggedIn)
  const [showPaymentModal, setShowPaymentModal] = useState(false)
  const [stripeClientSecret, setStripeClientSecret] = useState<string>('')
  const [stripePaymentAmount, setStripePaymentAmount] = useState<number>(null)
  const { width: windowWidth } = useContext(WindowDimensionsContext)

  const showPaginated =
    windowWidth - 100 < getBracketWidth(getNumRounds(bracket?.numTeams))

  const canPrint = bracket?.isPrintable
  const canSubmit = bracket?.isOpen && props.isUserLoggedIn
  const playStorage = new PlayStorage('loadStoredPicks', 'wpbb_play_data_')
  const paymentRequired = bracket?.isOpen && !userCanPlayBracketForFree // TODO: decide if we need to check if the bracket is open. Is this handled by canSubmit?
  const loginRedirectUrl =
    props.loginUrl +
    (props.bracket?.url ? `?redirect=${props.bracket.url}` : '')

  const clearError = () => {
    setAddToApparelError(false)
    setSubmitPicksError(false)
  }

  const setMatchTreeAndSaveInStorage = (tree: MatchTree) => {
    setMatchTree(tree)
    playStorage.storePlay(
      {
        bracketId: bracket?.id,
        picks: tree.toMatchPicks(),
        id: playStorage.loadPlay(bracket?.id)?.id,
      },
      bracket?.id
    )
  }

  const shouldNotCreateNewPlay = (playReq: PlayReq) => {
    return (
      storedPlay?.id &&
      JSON.stringify(storedPlay?.picks) === JSON.stringify(playReq.picks)
    )
  }

  const getPlayReq = () => {
    const picks = matchTree?.toMatchPicks()
    const bracketId = bracket?.id
    if (!picks || !bracketId) {
      const msg = 'Cannot create play. Missing picks'
      logger.error(msg)
      return
    }
    const playReq: PlayReq = {
      title: bracket?.title,
      bracketId: bracketId,
      picks: picks,
    }
    return playReq
  }

  const handleApparelClick = async () => {
    clearError()
    setProcessingAddToApparel(true)
    const playReq = getPlayReq()
    playReq.generateImages = true
    try {
      if (shouldNotCreateNewPlay(playReq)) {
        // regenerate play images anyway (in case they don't exist)
        await bracketApi.generatePlayImages(storedPlay.id)
      } else {
        const res = await bracketApi.createPlay(playReq)
        const playId = res.id
        const newReq = {
          ...playReq,
          id: playId,
        }
        playStorage.storePlay(newReq, bracket?.id)
      }
      window.location.assign(bracketProductArchiveUrl)
    } catch (err) {
      setProcessingAddToApparel(false)
      setAddToApparelError(true)
      logger.error(err)
    }
  }

  const handleSubmitPicksClick = async () => {
    clearError()
    setProcessingSubmitPicks(true)
    const playReq = getPlayReq()
    playReq.generateImages = false
    playReq.createStripePaymentIntent = paymentRequired

    if (shouldNotCreateNewPlay(playReq)) {
      if (!paymentRequired) {
        if (isUserLoggedIn) {
          window.location.assign(myPlayHistoryUrl)
        } else {
          setShowRegisterModal(true)
        }
        return
      }
      if (stripeClientSecret) {
        setShowPaymentModal(true)
        return
      }
      await bracketApi
        .createStripePaymentIntent({ playId: storedPlay.id })
        .then((res) => {
          setStripeClientSecret(res.clientSecret)
          setStripePaymentAmount(res.amount)
          setProcessingSubmitPicks(false)
          setShowPaymentModal(true)
        })
        .catch((err) => {
          setProcessingSubmitPicks(false)
          setSubmitPicksError(true)
          logger.error(err)
        })

      return
    }
    return bracketApi
      .createPlay(playReq)
      .then((res) => {
        const playId = res.id
        const newReq = {
          ...playReq,
          id: playId,
        }
        playStorage.storePlay(newReq, bracket?.id)
        setStoredPlay(newReq)
        if (paymentRequired) {
          setStripeClientSecret(res.stripePaymentIntentClientSecret)
          setStripePaymentAmount(res.stripePaymentAmount)
          setShowPaymentModal(true)
          setProcessingSubmitPicks(false)
        } else if (isUserLoggedIn) {
          window.location.assign(myPlayHistoryUrl)
        } else {
          setShowRegisterModal(true)
        }
      })
      .catch((err) => {
        setProcessingSubmitPicks(false)
        logger.error(err)
      })
  }

  const playBuilderProps = {
    matchTree,
    setMatchTree:
      canPrint || canSubmit ? setMatchTreeAndSaveInStorage : undefined,
    handleApparelClick: canPrint ? handleApparelClick : undefined,
    handleSubmitPicksClick: canSubmit ? handleSubmitPicksClick : undefined,
    canPlay: canPrint || canSubmit,
    processingAddToApparel,
    processingSubmitPicks,
    addToApparelError,
    submitPicksError,
    bracketMeta,
    setBracketMeta,
  }

  return (
    <>
      <SubmitPicksRegisterModal
        show={showRegisterModal}
        setShow={setShowRegisterModal}
        loginUrl={loginRedirectUrl}
      />
      <StripePaymentModal
        title={'Submit Your Picks'}
        show={showPaymentModal}
        setShow={setShowPaymentModal}
        clientSecret={stripeClientSecret}
        paymentAmount={stripePaymentAmount}
        myPlayHistoryUrl={myPlayHistoryUrl}
      />
      {showPaginated ? (
        <PaginatedPlayBuilder {...playBuilderProps} />
      ) : (
        <PlayBuilder {...playBuilderProps} />
      )}
    </>
  )
}

const WrappedPlayBuilderPage = WithWindowDimensions(
  WithMatchTree(PlayBuilderPage)
)

export default WrappedPlayBuilderPage
