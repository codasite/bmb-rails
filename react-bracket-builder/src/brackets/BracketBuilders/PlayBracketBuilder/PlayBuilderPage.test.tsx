import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import PlayBuilderPage from './PlayBuilderPage'
import { MatchTree } from '../../shared/models/MatchTree'
import '@testing-library/jest-dom/jest-globals'
import { PlayStorage } from '../../shared/storages/PlayStorage'
import { MatchPick, MatchRes } from '../../shared/api/types/bracket'
import { jest } from '@jest/globals'
import { bracketApi } from '../../shared/api/bracketApi'

jest.mock('react-lineto', () => {
  return {
    __esModule: true,
    default: () => {
      return <div />
    },
    SteppedLineTo: () => {
      return <div className={'mock-react-lineto'} />
    },
  }
})

jest.mock('../../shared/api/bracketApi')

describe('PlayBuilderPage', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })
  test('renders PlayBuilderPage correctly', () => {
    const { asFragment } = render(
      <PlayBuilderPage matchTree={MatchTree.fromNumTeams(10)} />
    )
    expect(asFragment()).toMatchSnapshot()
  })
  test('renders PlayBuilderPage from sessionStorage', () => {
    const playStorage = new PlayStorage('loadStoredPicks', 'wpbb_play_data_')
    const matches: MatchRes[] = [
      {
        id: 9,
        roundIndex: 0,
        matchIndex: 0,
        team1: { id: 17, name: 'Team 1' },
        team2: { id: 18, name: 'Team 2' },
      },
      {
        id: 10,
        roundIndex: 0,
        matchIndex: 1,
        team1: { id: 19, name: 'Team 3' },
        team2: { id: 20, name: 'Team 4' },
      },
      { roundIndex: 1, matchIndex: 1 },
    ]
    const picks: MatchPick[] = [
      { roundIndex: 0, matchIndex: 0, winningTeamId: 17 },
      { roundIndex: 0, matchIndex: 1, winningTeamId: 19 },
      { roundIndex: 1, matchIndex: 1, winningTeamId: 19 },
    ]

    playStorage.storePlay(
      {
        bracketId: 1,
        picks: picks,
      },
      1
    )
    expect(window.location.search).toContain('loadStoredPicks=true')
    expect(sessionStorage.getItem('wpbb_play_data_1')).toBeTruthy()
    const { asFragment: asFragmentSession } = render(
      <PlayBuilderPage bracket={{ id: 1, numTeams: 4, matches: matches }} />
    )
    const { asFragment } = render(
      <PlayBuilderPage matchTree={MatchTree.fromPicks(4, matches, picks)} />
    )
    expect(asFragmentSession()).toEqual(asFragment())
  })
  test('click add to apparel button', async () => {
    jest.mock('../../shared/storages/PlayStorage')
    PlayStorage.prototype.storePlay = jest.fn()
    const createPlayMock = bracketApi.createPlay as jest.MockedFunction<
      typeof bracketApi.createPlay
    >
    const playResMock = {
      bracketId: 1,
      id: 2,
      picks: [],
      title: 'title',
      url: 'url',
      status: 'status',
      author: 3,
      authorDisplayName: 'authorDisplayName',
      publishedDate: {
        date: 'date',
        timezone_type: 1,
        timezone: 'timezone',
      },
    }
    createPlayMock.mockResolvedValue(playResMock)

    const matches = [
      [
        {
          id: 9,
          roundIndex: 0,
          matchIndex: 0,
          team1: { id: 17, name: 'Team 1' },
          team2: { id: 18, name: 'Team 2' },
          team1Wins: true,
        },
        {
          id: 10,
          roundIndex: 0,
          matchIndex: 1,
          team1: { id: 19, name: 'Team 3' },
          team2: { id: 20, name: 'Team 4' },
          team1Wins: true,
        },
      ],
      [{ roundIndex: 1, matchIndex: 1, team2Wins: true }],
    ]
    const matchTree = MatchTree.deserialize({ rounds: matches })
    const { asFragment } = render(
      <PlayBuilderPage
        matchTree={matchTree}
        bracket={{ id: 1 }}
        bracketProductArchiveUrl="#"
        myPlayHistoryUrl="#"
      />
    )
    expect(screen.getByText('Add to Apparel')).toBeEnabled()
    const location = window.location
    delete window.location
    window.location = { assign: jest.fn() as any } as Location
    await userEvent.click(screen.getByText('Add to Apparel'))
    window.location = location
    expect(screen.getByText('Generating your bracket...')).toBeVisible()
  })
})
