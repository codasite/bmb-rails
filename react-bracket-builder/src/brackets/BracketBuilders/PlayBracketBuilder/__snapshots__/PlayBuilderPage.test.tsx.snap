// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayBuilderPage renders PlayBuilderPage correctly 1`] = `
<DocumentFragment>
  <div
    class="wpbb-reset tw-uppercase tw-bg-no-repeat tw-bg-top tw-bg-cover tw-dark"
    style="background-image: url(undefined);"
  >
    <div
      class="tw-flex tw-flex-col tw-items-center tw-max-w-screen-lg tw-m-auto"
    >
      <div
        class="tw-h-[140px] tw-flex tw-flex-col tw-justify-center tw-items-center"
      >
        <div
          class="tw-flex tw-items-center tw-justify-center tw-font-600 tw-gap-14"
        >
          <span
            class="tw-text-dd-blue dark:tw-text-white"
          >
            Theme
          </span>
          <button
            class="tw-flex tw-items-center tw-justify-end dark:tw-justify-start tw-w-[71px] tw-h-30 tw-px-2 tw-rounded-16 dark:tw-border-2 tw-border-solid tw-border-white tw-cursor-pointer tw-bg-dd-blue dark:tw-bg-none"
          >
            <div
              class="tw-w-[47px] tw-h-[22px] tw-rounded-16 tw-bg-white tw-text-10 tw-flex tw-items-center tw-justify-center"
            >
              <span
                class="tw-text-dd-blue tw-font-600 tw-text-sans tw-uppercase"
              >
                dark
              </span>
            </div>
          </button>
        </div>
      </div>
      <div
        class="tw-flex tw-flex-col tw-dark wpbb-default-bracket tw-relative"
      >
        <div
          class="tw-flex tw-flex-col tw-justify-end"
          style="margin-bottom: 95px; min-height: 0;"
        >
          <div
            class="tw-flex tw-flex-col tw-gap-24 tw-items-center"
          >
            <div
              class="wpbb-team-3-0-winner tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[297px] tw-h-[52px] tw-text-white tw-font-700 tw-bg-transparent tw-border-solid tw-border-white/50"
              style="border-width: 2px;"
            />
          </div>
        </div>
        <div
          class="tw-flex tw-flex-col tw-justify-center tw-h-100"
        >
          <div
            class="tw-flex tw-justify-between tw-relative tw-w-[985px]"
          >
            <div
              class="tw-flex tw-flex-col tw-justify-center tw-gap-[80px]"
            >
              <div
                class="tw-flex tw-flex-col tw-gap-[20px] tw-translate-y-[0px]"
              >
                <div
                  class="wpbb-team-0-0-left tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-white tw-font-500 tw-bg-transparent tw-border-solid tw-border-white/50"
                  style="border-width: 2px;"
                />
                <div
                  class="wpbb-team-0-0-right tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-white tw-font-500 tw-bg-transparent tw-border-solid tw-border-white/50"
                  style="border-width: 2px;"
                />
              </div>
              <div
                class="tw-h-[76px] tw-w-[115px]"
              />
              <div
                class="tw-h-[76px] tw-w-[115px]"
              />
              <div
                class="tw-h-[76px] tw-w-[115px]"
              />
            </div>
            <div
              class="tw-flex tw-flex-col tw-justify-center tw-gap-[236px]"
            >
              <div
                class="tw-flex tw-flex-col tw-gap-[20px] tw-translate-y-[0px]"
              >
                <div
                  class="wpbb-team-1-0-left tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-white tw-font-500 tw-bg-transparent tw-border-solid tw-border-white/50"
                  style="border-width: 2px;"
                />
                <div
                  class="wpbb-team-1-0-right tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-white tw-font-500 tw-bg-transparent tw-border-solid tw-border-white/50"
                  style="border-width: 2px;"
                />
              </div>
              <div
                class="tw-flex tw-flex-col tw-gap-[20px] tw-translate-y-[0px]"
              >
                <div
                  class="wpbb-team-1-1-left tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-white tw-font-500 tw-bg-transparent tw-border-solid tw-border-white/50"
                  style="border-width: 2px;"
                />
                <div
                  class="wpbb-team-1-1-right tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-white tw-font-500 tw-bg-transparent tw-border-solid tw-border-white/50"
                  style="border-width: 2px;"
                />
              </div>
            </div>
            <div
              class="tw-flex tw-flex-col tw-justify-center tw-gap-[548px]"
            >
              <div
                class="tw-flex tw-flex-col tw-gap-[20px] tw-translate-y-[0px]"
              >
                <div
                  class="wpbb-team-2-0-left tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-white tw-font-500 tw-bg-transparent tw-border-solid tw-border-white/50"
                  style="border-width: 2px;"
                />
                <div
                  class="wpbb-team-2-0-right tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-white tw-font-500 tw-bg-transparent tw-border-solid tw-border-white/50"
                  style="border-width: 2px;"
                />
              </div>
            </div>
            <div
              class="tw-flex tw-flex-col tw-justify-center tw-gap-[1172px]"
            >
              <div
                class="tw-flex tw-flex-col tw-gap-[20px] tw-translate-y-[-48px]"
              >
                <div
                  class="wpbb-team-3-0-left tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-white tw-font-500 tw-bg-transparent tw-border-solid tw-border-white/50"
                  style="border-width: 2px;"
                />
                <div
                  class="wpbb-team-3-0-right tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-white tw-font-500 tw-bg-transparent tw-border-solid tw-border-white/50"
                  style="border-width: 2px;"
                />
              </div>
            </div>
            <div
              class="tw-flex tw-flex-col tw-justify-center tw-gap-[548px]"
            >
              <div
                class="tw-flex tw-flex-col tw-gap-[20px] tw-translate-y-[0px]"
              >
                <div
                  class="wpbb-team-2-1-left tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-white tw-font-500 tw-bg-transparent tw-border-solid tw-border-white/50"
                  style="border-width: 2px;"
                />
                <div
                  class="wpbb-team-2-1-right tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-white tw-font-500 tw-bg-transparent tw-border-solid tw-border-white/50"
                  style="border-width: 2px;"
                />
              </div>
            </div>
            <div
              class="tw-flex tw-flex-col tw-justify-center tw-gap-[236px]"
            >
              <div
                class="tw-flex tw-flex-col tw-gap-[20px] tw-translate-y-[0px]"
              >
                <div
                  class="wpbb-team-1-2-left tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-white tw-font-500 tw-bg-transparent tw-border-solid tw-border-white/50"
                  style="border-width: 2px;"
                />
                <div
                  class="wpbb-team-1-2-right tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-white tw-font-500 tw-bg-transparent tw-border-solid tw-border-white/50"
                  style="border-width: 2px;"
                />
              </div>
              <div
                class="tw-flex tw-flex-col tw-gap-[20px] tw-translate-y-[0px]"
              >
                <div
                  class="wpbb-team-1-3-left tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-white tw-font-500 tw-bg-transparent tw-border-solid tw-border-white/50"
                  style="border-width: 2px;"
                />
                <div
                  class="wpbb-team-1-3-right tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-white tw-font-500 tw-bg-transparent tw-border-solid tw-border-white/50"
                  style="border-width: 2px;"
                />
              </div>
            </div>
            <div
              class="tw-flex tw-flex-col tw-justify-center tw-gap-[80px]"
            >
              <div
                class="tw-flex tw-flex-col tw-gap-[20px] tw-translate-y-[0px]"
              >
                <div
                  class="wpbb-team-0-4-left tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-white tw-font-500 tw-bg-transparent tw-border-solid tw-border-white/50"
                  style="border-width: 2px;"
                />
                <div
                  class="wpbb-team-0-4-right tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-white tw-font-500 tw-bg-transparent tw-border-solid tw-border-white/50"
                  style="border-width: 2px;"
                />
              </div>
              <div
                class="tw-h-[76px] tw-w-[115px]"
              />
              <div
                class="tw-h-[76px] tw-w-[115px]"
              />
              <div
                class="tw-h-[76px] tw-w-[115px]"
              />
            </div>
          </div>
        </div>
        <div
          style="margin-top: 20px; min-height: 200px;"
        >
          <div
            class="tw-flex tw-flex-col tw-gap-20 tw-justify-between tw-items-center tw-font-700 tw-whitespace-nowrap "
          >
            <span
              class="tw-text-36 tw-text-dd-blue dark:tw-text-white"
            >
              Who You Got?
            </span>
            <svg
              class="tw-w-[124px] tw-text-black/25 dark:tw-text-white"
            />
          </div>
        </div>
        <div
          class="tw-absolute"
        >
          <div
            class="mock-react-lineto"
          />
          <div
            class="mock-react-lineto"
          />
          <div
            class="mock-react-lineto"
          />
          <div
            class="mock-react-lineto"
          />
          <div
            class="mock-react-lineto"
          />
          <div
            class="mock-react-lineto"
          />
          <div
            class="mock-react-lineto"
          />
          <div
            class="mock-react-lineto"
          />
          <div
            class="mock-react-lineto"
          />
          <div
            class="mock-react-lineto"
          />
          <div
            class="mock-react-lineto"
          />
          <div
            class="mock-react-lineto"
          />
          <div
            class="mock-react-lineto"
          />
          <div
            class="mock-react-lineto"
          />
          <div
            class="mock-react-lineto"
          />
          <div
            class="mock-react-lineto"
          />
        </div>
        <div
          class="tw-absolute"
        >
          <div />
          <div />
        </div>
      </div>
      <div
        class="tw-px-24 tw-mt-60 tw-flex tw-gap-15 tw-flex-col tw-items-stretch tw-self-stretch"
      >
        <div
          class="tw-flex tw-self-stretch tw-flex-col tw-items-stretch"
        >
          <button
            class="tw-flex tw-flex-row tw-items-center tw-justify-center tw-font-sans tw-uppercase tw-whitespace-nowrap tw-bg-transparent tw-text-white/20 tw-gap-10 tw-text-24 tw-font-700 tw-rounded-8 tw-py-16 tw-border-solid tw-border tw-border-white/20"
            disabled=""
          >
            <svg
              style="height: 24px;"
            />
            <span>
              Add to Apparel
            </span>
          </button>
        </div>
        <button
          class="tw-flex tw-flex-row tw-items-center tw-justify-center tw-font-sans tw-uppercase tw-whitespace-nowrap tw-bg-transparent tw-text-white/20 tw-gap-10 tw-text-24 tw-font-700 tw-rounded-8 tw-py-16 tw-border-solid tw-border tw-border-white/20"
          disabled=""
        >
          <svg
            style="height: 24px;"
          />
          Submit picks
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
